exporting base history events csv (#6633)
rotki/rotki: docs/api.rst

   :statuscode 409: No user is logged in or one of the identifiers to delete did not correspond to an event in the DB or one of the identifiers was for the last event in the corresponding transaction hash and force_delete was false..
   :statuscode 500: Internal rotki error

> .. _filter-request-args-label:

Reviewer: This shouldn't be in the middle of all endpoints. It should be at the very end. Also did you try to do `make html` and look at the generated docs to make sure the link works fine and renders fine?

> Filter Request Arguments

Reviewer: ```suggestion
HistoryEvents Filter Request Arguments
```

------------------------

> These request arguments are common to multiple parts of the documentation:

Reviewer: ```suggestion
These filter request arguments are common to multiple api endpoints that deal with history events filtering.
```

- :reqjson list[string] products: An optional list of product type to filter by. List of strings. This will make it an EVM event query.
- :reqjson list[string] addresses: An optional list of EVM addresses to filter by in the set of counterparty addresses. This will make it an EVM event query.
- :reqjson list[int] validator_indices: An optional list of validator indices to filter by. This makes it an EthStakingevent query.

Exporting History Events
============================================

.. http:post:: /api/(version)/history/events/export

   Doing a POST on this endpoint with the given filter parameters will export a csv with all history events matching the filter to a file in the provided directory. Only the 'directory_path' argument is required. If no filter is used all the events will be exported.

   **Example Request**:

   .. http:example:: curl wget httpie python-requests

      POST /api/1/history/events/export HTTP/1.1
      Host: localhost:5042
      Content-Type: application/json;charset=UTF-8

      {
          "directory_path": "/home",
          "from_timestamp": 1500,
          "to_timestamp": 999999
      }

>    .. _history_export_schema_section:

Reviewer: you should also use the link in the other endpoints where you saw them. Not just in the new endpoints.

Author: so the set of filters we use here were only used in these two

   :statuscode 409: No user is logged in or one of the identifiers to delete did not correspond to an event in the DB or one of the identifiers was for the last event in the corresponding transaction hash and force_delete was false..
   :statuscode 500: Internal rotki error

Exporting History Events
============================================

> .. http:post:: /api/(version)/history/events/export

Author: combining both, from what Lefteris did here, they have the same request format, it's different for me

   :statuscode 409: No user is logged in or one of the identifiers to delete did not correspond to an event in the DB or one of the identifiers was for the last event in the corresponding transaction hash and force_delete was false..
   :statuscode 500: Internal rotki error

Exporting History Events
============================================

.. http:post:: /api/(version)/history/events/export

   Doing a POST on this endpoint with the given filter parameters will return a csv with all history events matching the filter. All arguments are optional. If nothing is given all events will be returned.

> 

Author: done

> 

Reviewer 2: now all the arguments are not optional since you have the directory path

  :statuscode 200: Reprocessing went fine.
  :statuscode 409: An issue ocurred during reprocessing
  :statuscode 500: Internal rotki error

> Schema

Reviewer: nitpick title change for later

```suggestion
Schema Appendix
```
